packages/design-preview/src/components/style-variation.tsx

```diff
@@ -1,9 +1,12 @@
+import { isEnabled } from '@automattic/calypso-config';
 import { GlobalStylesContext } from '@wordpress/edit-site/build-module/components/global-styles/context';
 import { mergeBaseAndUserConfigs } from '@wordpress/edit-site/build-module/components/global-styles/global-styles-provider';
 import Preview from '@wordpress/edit-site/build-module/components/global-styles/preview';
 import { useMemo } from '@wordpress/element';
 import classnames from 'classnames';
 import { translate } from 'i18n-calypso';
+// eslint-disable-next-line no-restricted-imports
+import PremiumBadge from 'calypso/components/premium-badge';
```

Comment:
Hi there! As a rule of thumb, we avoid importing from `calypso/*` in packages to avoid possible circular dependencies. See: https://github.com/Automattic/wp-calypso/blob/trunk/.eslintrc.js#L57

Reply:
Hi @taipeicoder, is there a recommended approach for using Calypso components in packages?

I briefly wondered if we should have moved this badge into its own package, but then again, it internally has 2 Calypso imports, so we'd just be moving the problem. 🤔

Reply:
A possible workaround is to pass the badge as a prop to the component, this way `<PremiumBadge />` can be imported in the Stepper component and used in the package 🙂

Reply:
I think that overcomplicates the implementation because we need to import a component in a totally unrelated component that needs to be passed down multiple levels until we reach the component that really needs to render it.

IMO, it should be fine to disable the `no-restricted-imports` here since we're sure the premium badge component won't introduce any circular dependency. That's kind of [what we've done in other packages](https://github.com/search?q=no-restricted-imports+repo%3AAutomattic%2Fwp-calypso+path%3Apackages&type=Code&ref=advsearch&l=&l=).

Reply:
FWIW, the `@automattic/design-picker` package (which is the one modified here), already imports stuff from Calypso:

https://github.com/Automattic/wp-calypso/blob/cfe04e1513af8d47f7f34dc9fae861af7f11aa8a/packages/design-picker/src/hooks/use-theme-designs-query.ts#L2

https://github.com/Automattic/wp-calypso/blob/04cd987b98fe52c942bc6bf1165373a9b9e47a8d/packages/design-picker/src/components/design-picker-category-filter/unified-design-picker-category-filter.tsx#L2

Reply:
Yep, not ideal. In any case, my personal preference is that `packages/*` should be as isolated as possible so my approach would either be passing PremiumBadge as a prop (which as you mentioned, creates a lot of passing down props), or move PremiumBadge to `@automattic/design-picker` or `@automattic/components`.

I don't see this as a blocker though, as long as it doesn't create bundle or module dependency issues 👍

Reply:
> move PremiumBadge to @automattic/design-picker or @automattic/components.

Unfortunately, that wouldn't avoid the external dependency, since `PremiumBadge` requires `calypso/components/badge` and `calypso/lib/touch-detect`. 

> passing PremiumBadge as a prop (which as you mentioned, creates a lot of passing down props)

Finally opted for this to help moving things forward: https://github.com/Automattic/wp-calypso/pull/69082/commits/9bbf5bc272ae282be738962f366f318c2bd8ad3c